CriticalMass/SettingsViewController.swift

```diff
@@ -69,8 +69,34 @@ class SettingsViewController: UITableViewController {
             let name = String(describing: cell)
             tableView.register(UINib(nibName: name, bundle: nil), forCellReuseIdentifier: name)
         }
+        configureSettingsFooter()
         configureNavigationBar()
     }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        guard let footerView = tableView.tableFooterView else {
+            return
+        }
+        let height = footerView.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
+        var footerFrame = footerView.frame
+        if height != footerFrame.size.height {
+            footerFrame.size.height = height
+            footerView.frame = footerFrame
+            tableView.tableFooterView = footerView
```

Comment:
Why do we need to set the tableView.tableFooterView again? And ist the tableView.layoutIfNeeded() needed? I’m afraid that we could end up in endless layout cycles by triggering a layoutPass in viewDidLayoutSubviews().

Reply:
It is only being set if the height actually has changed. So a new layout cycle is only triggered once. Better explained [here](https://useyourloaf.com/blog/variable-height-table-view-header/)

Reply:
These lines are working for me or am I missing something? 
```swift
guard let footerView = tableView.tableFooterView else {
        return
}
let height = footerView.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
footerView.frame.size.height = height
```

Reply:
That doesn't produce a footer that is too big?

Reply:
yes

Reply:
Ok. I'll change it to the version you pointed out

Reply:
Had to add the height check though so that we do not set the height every layout cycle